### PR TITLE
Add the system struct of GRFFE system

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

Dependent on 
- [x] #4435 

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/ForceFree/System.hpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ * \f{align*}{
+ *  \nabla_a F^{ab} &= -J^b,\\
+ *  \nabla_a ^* F^{ab} &= 0.
+ * \f}
+ *
+ * \note We are using the electromagnetic variables with the scaling convention
+ * that the factor \f$4\pi\f$ does not appear in Maxwell equations and the
+ * stress-energy tensor of the EM fields.
+ *
+ * where \f$F^{ab}\f$ is the electromagnetic field tensor, \f$^*F^{ab}\f$ is its
```
Comment (nilsdeppe) [inline src/Evolution/Systems/ForceFree/System.hpp:38]: move up above note?

File: src/Evolution/Systems/ForceFree/System.hpp [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+ * electrodynamics (GRFFE) assumes the following which are called the force-free
+ * (FF) conditions.
+ *
+ * \f{align*}{
+ *  F^{ab}J_b & = 0, \\
+ *  ^*F^{ab}F_{ab} & = 0, \\
+ *  F^{ab}F_{ab} & > 0.
+ * \f}
+ *
+ * There are a number of different ways in literature to numerically treat the
```
Comment (nilsdeppe) [inline src/Evolution/Systems/ForceFree/System.hpp:149]: is there a paper that could be cited here?

File: src/Evolution/Systems/ForceFree/System.hpp [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+ * \f}
+ *
+ * See the documentation of Fluxes and Sources for further details.
+ *
+ * In addition to Maxwell equations, general relativistic force-free
+ * electrodynamics (GRFFE) assumes the following which are called the force-free
+ * (FF) conditions.
+ *
+ * \f{align*}{
+ *  F^{ab}J_b & = 0, \\
```
Comment (nilsdeppe) [inline src/Evolution/Systems/ForceFree/System.hpp:132]: Could you expand these to also be in terms of E & B?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash immediately :)

File: src/Evolution/Systems/ForceFree/System.hpp (hunk 2 above)
Comment (yoonso0-0) [reply src/Evolution/Systems/ForceFree/System.hpp:149]: I'll do some quick research and add papers

File: src/Evolution/Systems/ForceFree/System.hpp (hunk 3 above)
Comment (yoonso0-0) [reply src/Evolution/Systems/ForceFree/System.hpp:132]: Sure

File: src/Evolution/Systems/ForceFree/System.hpp (hunk 2 above)
Comment (ermost) [reply src/Evolution/Systems/ForceFree/System.hpp:149]: There is some recent discussion in this paper: https://arxiv.org/pdf/2007.06580.pdf

File: src/Evolution/Systems/ForceFree/System.hpp (hunk 3 above)
Comment (ermost) [reply src/Evolution/Systems/ForceFree/System.hpp:132]: E_i B^i =0

E_i E^i - B_i B^i <0.

Maybe state F.J=0 directly in terms of the drift current?

**Comment by yoonso0-0** [comment]
Fixed things on comments, added some references and mentioned their techniques on FF constraints :)

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**